### PR TITLE
Added function to access sensor resourcestatus

### PR DESCRIPTION
**Pull request by bigblueswope**
resourcestatus API endpoint provides access to resource utilization data for each sensor.  Facilitates monitoring of sensors to look for sensors that may be experiencing high resource utilization.

### CODE REVIEW

File: src/cbapi/legacy/cbapi.py [hunk 1]
```diff
@@ -294,6 +294,14 @@ def sensor(self, sensor_id):
         r.raise_for_status()
         return r.json()
 
+    def sensor_resourcestatus(self, sensor_id):
+        """
+        get resourve utilization for a single sensor, as specified by sensor id
```
Comment (jgarman) [inline src/cbapi/legacy/cbapi.py]: fix misspelling ("resourve") by committing a change in your branch

File: src/cbapi/legacy/cbapi.py [hunk 2]
```diff
@@ -294,6 +294,14 @@ def sensor(self, sensor_id):
...
+        """
+        r = self.cbapi_get("%s/api/v1/sensor/%s/resourcestatus" % (self.server, sensor_id))
+        r.raise_for_status()
+        return r.text
```
Comment (jgarman) [inline src/cbapi/legacy/cbapi.py]: is the output of the resourcestatus endpoint JSON? If so let's decode it from json here. I get 204 ("no content") on my cb server when calling this endpoint against any of my sensors.

**Comment by bigblueswope** [comment]
The returning r.json fails.  Returning r.text that is then converted to JSON via json.loads() works.  I don't know why, but in my working with it that's the only thing that worked.

**Comment by bigblueswope** [comment]
I've noticed that not all of my sensors have that data available.  
screen shot 2016-05-22 at 11 50 36 am
screen shot 2016-05-22 at 11 51 35 am

**Comment by bigblueswope** [comment]
Ahhh. I just put it together... returning r.json() fails for the sensors that don't have the data but returning r.text() works because it doesn't have the json formatting requriements.  What would you suggest for differentiating between 204 and 200 response codes then?

**Comment by jgarman** [comment]
Yeah this is a problem for several endpoints since this behavior (returning 204 instead of an empty json object) was introduced in 5.1. Need a new function .decode_json() in the cbapi object which takes in the response and a default value. If the response status code is not 200, return the default value (in this case, None). If the status code is 200, return json.loads(response.text). Then change all the r.json() calls to this new decode_json() function.

**Comment by jgarman** [comment]
Cool I'll merge this and fix up the 204 mess Thanks!

**Pull request merged**